### PR TITLE
Implement verbose to ShapRSECV

### PR DESCRIPTION
**Pull request by Matgrb**
#21

### CODE REVIEW

File: probatus/feature_elimination/feature_elimination.py [hunk 1]
```diff
@@ -160,7 +170,7 @@ def _check_if_fitted(self):
 
 
     @staticmethod
-    def _preprocess_data(X):
+    def _preprocess_data(X, verbose=0):
```
Comment (timvink) [inline probatus/feature_elimination/feature_elimination.py:173]: this is a private method. Why not remove the @staticmethod and use `self.verbose`?

File: probatus/feature_elimination/feature_elimination.py [hunk 2]
```diff
@@ -430,14 +465,14 @@ def fit(self, X, y):
...
-                  f'Removed features at the end of the round: {features_to_remove}')
+            if self.verbose > 50:
+                print(f'Round: {round_number}, Current number of features: {len(current_features_set)}, '
+                      f'Current performance: Train {self.report_df.loc[round_number]["train_metric_mean"]} '
+                      f'+/- {self.report_df.loc[round_number]["train_metric_std"]}, CV Validation '
+                      f'{self.report_df.loc[round_number]["val_metric_mean"]} '
+                      f'+/- {self.report_df.loc[round_number]["val_metric_std"]}. \n'
+                      f'Num of features left: {len(remaining_features)}. '
+                      f'Removed features at the end of the round: {features_to_remove}')
         self.fitted = True
```
Comment (timvink) [inline probatus/feature_elimination/feature_elimination.py:476]: just noticed this. Remember if you have attributes with a trailing _ in the class, sklearn sees it as a fitted attribute. See this source code https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/utils/validation.py#L955

File: tests/feature_elimination/test_feature_elimination.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def test_shap_rfe_randomized_search(X, y):
     }
     search = RandomizedSearchCV(clf, param_grid, cv=2)
 
-    shap_elimination = ShapRFECV(search, step=0.8, cv=2, scoring='roc_auc', n_jobs=4)
+    shap_elimination = ShapRFECV(search, step=0.8, cv=2, scoring='roc_auc', n_jobs=4, verbose=150)
```
Comment (timvink) [inline tests/feature_elimination/test_feature_elimination.py:30]: ideally, write another unit test with verbose=0 and assert that `caplog` is empty.

**Review by timvink — COMMENTED** [review]
some minor remarks, feel free to merge already

File: probatus/feature_elimination/feature_elimination.py (hunk 1 above)
Comment (Matgrb) [reply probatus/feature_elimination/feature_elimination.py:173]: This function will be moved to utils in #7 . For now i will merge it like this.

File: probatus/feature_elimination/feature_elimination.py (hunk 2 above)
Comment (Matgrb) [reply probatus/feature_elimination/feature_elimination.py:476]: #24 will cover this

File: tests/feature_elimination/test_feature_elimination.py (hunk 3 above)
Comment (Matgrb) [reply tests/feature_elimination/test_feature_elimination.py]: Done
